src/frontend/src/utils/credential-devices.ts

```diff
@@ -13,8 +13,22 @@ const derFromPubkey = (pubkey: DeviceKey): DerEncodedPublicKey =>
 
 export const convertToCredentialData = (
   device: Omit<DeviceData, "alias">
-): CredentialData => ({
-  credentialId: Buffer.from(device.credential_id[0] ?? []),
-  pubkey: derFromPubkey(device.pubkey),
-  origin: device.origin[0],
-});
+): CredentialData | undefined => {
+  // In certain cases, e.g. Chrome on Windows 10, an invalid credential id is
+  // not ignored but instead will result in a WebAuthn error that prevents a
+  // user from authenticating with any of their registered devices.
+  //
+  // Instead of throwing an error, we return `undefined` to make sure that the
+  // device will be filtered out to unblock the user from authenticating.
+  if (
+    device.credential_id.length !== 1 ||
+    device.credential_id[0].length === 0
```

Comment:
Was this the error? To not check the value inside the `credential_id`?

Reply:
The current implementation falls back to an empty buffer when a device doesn't have a credential id:

https://github.com/dfinity/internet-identity/blob/680f8a0aca77de5bf6f97b33daf98b51a3f2ac7b/src/frontend/src/utils/credential-devices.ts#L17

Previously this never happened, since a flatmap would filter out devices without a credential id before they're converted into credential data:

https://github.com/dfinity/internet-identity/blob/5a685b03d58f79f189a0964e73970fdbcd3c7b2c/src/frontend/src/utils/iiConnection.ts#L212

Since the current implementation falls back to an empty buffer, the WebAuthn API on some devices will throw an exception instead of ignoring these in the `allowedCredentials` list.

The changes and added tests in this PR, attempt to guarantee that credential data will never contain an empty buffer as credential id.

Reply:
Thanks for the clarification!